### ISSUE TITLE
[Discussion]: Pawns keep shooting the person right in front of them

### ISSUE THREAD

**Issue by Wuldahfel** [CombatExtended-Continued/CombatExtended#581]
In vanilla, when pawns are only 1 tile away from your colonist's standing position, they don't shoot them. It makes sense, as realistically, either the person in front of them would be ducking or the one standing behind stands slightly to the side while shooting, using the person in front of them as a cover. It frustrates me every time insects pop up and I put my melee colonists in front reflexively, only to have the person behind them shoot them right in the head. It makes me unable to play permadeath.

I even downloaded the "Avoid Friendly Fire" mod to avoid them shooting when someone is in front of them but that mod also acknowledges the pawn in front of the colonist has no chance of getting hit, so it won't prevent them from shooting, so my colonists still shoot my melee guys right in the head from 2 centimeters behind them.

I know this is probably an issue with the new projectile based system but couldn't there be a possible adjustment for this? Like making the friendly pawns in front count as a wall to take cover behind somehow? Or I noticed that the colonists sometimes duck when they're behind cover, make the melee pawn do that when in this situation? I probably want too much but I just want my melee colonists to be useful without getting constantly assassinated by my own dudes.

**Comment by Wuldahfel** [comment]
I suck at this label thing don't I.

**Comment by LockdownX7** [comment]
> In vanilla, when pawns are only 1 tile away from your colonist's standing position, they don't shoot them.
This behaviour is in CE as well, and works as expected in ranged combat, but it does get a bit screwy when enemies get into melee range of the formation. I'll look into possible solutions.

**Comment by randomtyper** [comment]
Do you use the combat effects for CE mod?

**Comment by Wuldahfel** [comment]
I'm not using the combat effects mod, was thinking about using it but I saw in the comments that it screws up armor pen, so decided against it. I am using the "Combat Extended Fixes" mod though, but I don't think it messes with any projectile hediffs.

I just tested and yeah, surely enough it does work normally when shooting at a wall from a distance behind another colonist. But I clearly remember when I was fighting a big infestation for a mission, my melee colonist was standing in front of the doorway, and the ones standing right behind him just murdered him. I reloaded and later decided to just burn the entire infestation with incendiary launchers, which left ash and smoke everywhere and got everyone who wasn't wearing PA helmets downed from smoke inhalation, so I had to carry them outside and they had to sleep outside for almost two seasons until the smoke went away lol.

The quest's rewards were a good quality PA, an orbital power beam targeter and 142 uranium though, so it was well worth it.
